File: js/sort/bubble.js
/**
 * 冒泡排序
 */
var bubbleSort = function(arr) {
  var endIndex = arr.length - 1

  do {
    var hasSwapElement = false

    //单次比较
    for(var i = 0; i <= endIndex; i++) {
      if(arr[i] > arr[i + 1]) {
        var temp = arr[i]
        arr[i] = arr[i + 1]
        arr[i + 1] = temp
        hasSwapElement  = true
      }
    }

    endIndex = endIndex - 1
  } while (hasSwapElement)

  return arr
}

module.exports = bubbleSort

File: python/sort/bubble.py
def bubbleSort(arr):
    global swapFlag
    swapFlag = True
    lastIndex = len(arr)
    while swapFlag:
        swapFlag = False
        for i in range(0, lastIndex-1):
            if arr[i + 1] < arr[i]:
                tmp = arr[i]
                arr[i] = arr[i + 1]
                arr[i + 1] = tmp
                swapFlag = True
        lastIndex = lastIndex - 1
    return arr
